src/modules/audio/audio.c

```diff
@@ -440,4 +454,45 @@ struct SoundData* lovrAudioCapture(uint32_t frameCount, SoundData *soundData, ui
   }
 
   return soundData;
+}
+
+void lovrAudioGetDevices(AudioDevice **outDevices, size_t *outCount) {
+  if(state.deviceInfos) 
+    free(state.deviceInfos);
+  
+  ma_result gettingStatus = ma_context_get_devices(&state.context, NULL, NULL, NULL, NULL);
+  lovrAssert(gettingStatus == MA_SUCCESS, "Failed to enumerate audio devices: %d", gettingStatus);
+  *outCount = state.context.playbackDeviceInfoCount + state.context.captureDeviceInfoCount;
+  *outDevices = state.deviceInfos = calloc(*outCount, sizeof(AudioDevice));
+  for(int i = 0; i < *outCount; i++) {
+    ma_device_info *mainfo = &state.context.pDeviceInfos[i];
+    AudioDevice *lovrInfo = &state.deviceInfos[i];
+    lovrInfo->name = mainfo->name;
+    lovrInfo->type = i < state.context.playbackDeviceInfoCount ? AUDIO_PLAYBACK : AUDIO_CAPTURE;
+    lovrInfo->isDefault = mainfo->isDefault; // remove _private after bumping miniaudio
+    lovrInfo->identifier = &mainfo->id;
+    lovrInfo->minChannels = mainfo->minChannels;
+    lovrInfo->maxChannels = mainfo->maxChannels;
+  }
+}
+
+void lovrAudioUseDevice(AudioDeviceIdentifier identifier, int sampleRate, SampleFormat format) {
+  int deviceCount = state.context.playbackDeviceInfoCount + state.context.captureDeviceInfoCount;
+  for(int i = 0; i < deviceCount; i++) {
+    if (identifier == &state.context.pDeviceInfos[i].id) {
+      AudioType type = i < state.context.playbackDeviceInfoCount ? AUDIO_PLAYBACK : AUDIO_CAPTURE;
+      state.config[type].device = identifier;
+      if (sampleRate) state.config[type].sampleRate = sampleRate;
+      if (format != SAMPLE_INVALID) state.config[type].format = format;
+      lovrLog(LOG_INFO, "audio", "Switching to %s device %s (%p)", type?"capture":"playback", state.context.pDeviceInfos[i].name, identifier);
+      ma_device_uninit(&state.devices[type]);
+      if(state.config[type].enable)
+        lovrAudioInitDevice(type);
+      if(state.config[type].start)
+        ma_device_start(&state.devices[type]);
+
+      return;
+    }
+  }
+  lovrAssert(false, "Couldn't find the given identifier");
 }
```

Comment:
I'm using Visual Studio Code, I thought it would pick up editorconfig autiomatically, but apparently you need to install an extension for that?? doing that :S 

I have now enabled it on two of my computers but uh... I have a lot of computers :S